src/ClassicUO.IO/DataReader.cs

```diff
@@ -46,6 +46,8 @@ public unsafe class DataReader : IDisposable
         private byte* _data;
         private GCHandle _handle;
 
+        public bool HasData => _data != null;
```

Comment:
I don't think this check is correct. We should do an IntPtr.Zero check or use this:

https://learn.microsoft.com/en-us/dotnet/api/system.runtime.compilerservices.unsafe.isnullref?view=net-8.0

Reply:
Isn't that the same, just a different (and more complicated) way to do it?

> This check is conceptually similar to (void*)(&source) == nullptr.

.... but what does "similar" mean?

Reply:
note this compares the `void*`, not the `IntPtr`.

Reply:
I think the Microsoft documentation means that comparing `IntPtr` to `null` may implicitly convert the `IntPtr` to `IntPtr?` which is nullable; but this nulled value means the absence of an `IntPtr` value, which is different from `IntPtr.Zero` or just `0`.

But none of this applies to raw pointers. Comparing those to `null` is fine.

Reply:
Ah I see what you mean, the fact that it is a raw pointer, makes it valid to check against null vs a ref or a Intptr. Ok.